### PR TITLE
Add tests for Running statistics

### PR DESCRIPTION
**Pull request by rleonid**
More accurate calculation of mean when joining.

### CODE REVIEW

File: src/lib/running.ml [hunk 1]
```diff
@@ -63,9 +63,12 @@ let join rs1 rs2 =
...
        else let new_size = float (rs1.size + rs2.size) in
             let new_mean =
-              let num = rs1.mean *. (float rs1.size) 
+              let p1 = (float rs1.size) /. new_size
+              and p2 = (float rs2.size) /. new_size in
+              rs1.mean *. p1 +. rs2.mean *. p2
+              (* let num = rs1.mean *. (float rs1.size)
                       +. rs2.mean *. (float rs2.size) in
-              num /. new_size
+              num /. new_size *)
```
Comment (smondet) [inline src/lib/running.ml]: Why leave the commented code?

File: src/lib/running.mli [hunk 2]
```diff
@@ -13,10 +13,10 @@ type t = { size : int         (** Number of observations. *)
 val empty : t
 
 (** [init x] initialize a [t] with [x]. *)
-    val init : float -> t
+val init : float -> t
```
Comment (smondet) [inline src/lib/running.mli:16]: `[init x] initialize a [t] with [x]` → I would read <code>[init x] initialize<b>S</b></code>

**Comment by smondet** [comment]
LGTM, the comments are minor.

**Comment by rleonid** [comment]
Thanks Seb!

**Pull request merged**